scienceplots/tests/test_scienceplots.py

```
@@ -42,4 +42,50 @@ def test_usage_of_each_style(
+    with plt.style.context(['science', 'no-latex']):
+        assert plt.rcParams['text.usetex'] is False, \
+            "rcParams['text.usetex'] should be False when 'no-latex' is applied after 'science'."
+
+    # Final check: After the context, rcParams should be restored.
+    # If 'science' was the baseline and set usetex True, it should revert to that.
+    # If MPL default for usetex is False, and 'science' was not applied before this test's contexts,
+    # it should revert to False or its original state prior to this test.
+    # This part can be tricky as test order is not guaranteed and rcParams are global.
+    # The main goal is to test the override *within* the specific context.
+    # For robustness, let's check against the known state of 'science' if it was applied.
+    if science_sets_usetex:
+        # This assumes that exiting the context restores to the state *before that context*.
+        # If the outer state was 'science', it should return to text.usetex = True.
+        # If the outer state was default matplotlib, it would be text.usetex = False (usually).
+        # To be absolutely sure, one might need to snapshot rcParams, but that's overkill here.
+        # The critical assertion is the one *inside* the ['science', 'no-latex'] context.
+        # Let's re-apply 'science' to check its state for clarity if it was supposed to be True.
+        with plt.style.context('science'):
+            assert plt.rcParams['text.usetex'] is True, \
+                "rcParams['text.usetex'] should revert to True for 'science' style after other contexts."
+    else:
+        # If 'science' style itself doesn't set 'text.usetex':True (e.g. if science.mplstyle was modified),
+        # then check if it's False (Matplotlib default).
+        assert plt.rcParams['text.usetex'] is plt.style.library['default']['text.usetex'], \
+            "rcParams['text.usetex'] should revert to Matplotlib default if 'science' doesn't set it."
```

This feels awkwardly complicated for a test. Let it fail if `science.mplstyle` is modified - we just want to test things go as expected, if a test adapts to whatever is written in the file, then it loses it's purpose. The same applies to the prior code: assume science sets usetex.